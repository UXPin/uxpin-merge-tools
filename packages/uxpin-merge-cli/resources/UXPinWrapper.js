import React from "react";

export default function UXPinWrapper({ children }) {
  return children;
}
