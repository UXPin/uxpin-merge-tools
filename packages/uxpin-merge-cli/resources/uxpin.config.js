module.exports = {
  components: {
    categories: [
      {
        name: 'General',
        include: [
          // 'src/components/Button/Button.js',
        ]
      }
    ],
    wrapper: 'src/Wrapper/UXPinWrapper.js',
    webpackConfig: 'uxpin.webpack.config.js',
  },
  name: 'Example Design System'
};
