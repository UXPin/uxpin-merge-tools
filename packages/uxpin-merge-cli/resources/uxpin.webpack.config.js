const path = require("path");
const webpack = require("webpack");

module.exports = {
  output: {
    path: path.resolve(__dirname, "build"),
    filename: "bundle.js",
    publicPath: "/"
  },
  resolve: {
    modules: [__dirname, "node_modules"],
    extensions: ["*", ".js", ".jsx"]
  },
  devtool: "source-map",
  module: {
    rules: [
      {
        test: /\.(s*)css$/,
        use: [
          {
            loader: 'style-loader'
          },
          {
            loader: 'css-loader',
            options: {
              importLoaders: 2
            }
          },
        ]
      },
      {
        loader: "babel-loader",
        test: /\.js?$/,
        exclude: /node_modules/,
        options: {
          presets: ['@babel/preset-env', '@babel/preset-react'],
        }
      },
    ]
  }
}
