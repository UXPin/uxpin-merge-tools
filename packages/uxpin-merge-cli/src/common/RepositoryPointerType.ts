export const enum RepositoryPointerType {
  Branch = 'branch',
  Tag = 'tag',
}
