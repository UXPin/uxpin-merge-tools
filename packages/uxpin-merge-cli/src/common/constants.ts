export const DEFAULT_BRANCH_NAME = 'master';

export const HEAD_REF_NAME = 'HEAD';

export const SHORT_COMMIT_HASH_IDX = 7;
