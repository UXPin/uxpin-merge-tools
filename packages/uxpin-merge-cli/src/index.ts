import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
