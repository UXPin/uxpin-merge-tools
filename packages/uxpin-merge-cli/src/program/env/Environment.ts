export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
