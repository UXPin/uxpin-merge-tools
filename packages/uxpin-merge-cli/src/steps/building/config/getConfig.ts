import { join } from 'path';
import { Configuration } from 'webpack';
import { smart } from 'webpack-merge';
import { BuildOptions } from '../BuildOptions';
import { getComponentLibraryInputPath } from '../library/getComponentLibraryInputPath';

export const DEBUG_APP_BUNDLED_FILE = 'index.js';
export const TEMP_DIR_NAME = '.uxpin-merge';
export const TEMP_DIR_PATH = `./${TEMP_DIR_NAME}`;
export const LIBRARY_INPUT_FILENAME = `components.js`;
export const LIBRARY_OUTPUT_FILENAME = 'designsystemlibrary.js';

export function getConfig({ development, webpackConfigPath, projectRoot, uxpinDirPath }: BuildOptions): Configuration {
  const config: Configuration = decorateWithDevToolsWhenDevelopment(
    {
      entry: getComponentLibraryInputPath(uxpinDirPath),
      mode: development ? 'development' : 'production',
      optimization: {
        runtimeChunk: false,
        splitChunks: false,
      },
      output: {
        filename: LIBRARY_OUTPUT_FILENAME,
        libraryTarget: 'commonjs',
        path: uxpinDirPath,
      },
      resolve: {
        extensions: ['.js', '.jsx'],
      },
    },
    development
  );

  if (webpackConfigPath) {
    const userWebpackConfig: Configuration = require(join(projectRoot, webpackConfigPath));
    return smart(userWebpackConfig, config);
  }

  return config;
}

function decorateWithDevToolsWhenDevelopment(config: Configuration, development = false): Configuration {
  if (!development) {
    return config;
  }

  return {
    ...config,
    devtool: 'cheap-module-eval-source-map',
  };
}
