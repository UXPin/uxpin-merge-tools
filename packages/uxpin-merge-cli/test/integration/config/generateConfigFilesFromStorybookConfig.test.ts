import { emptyDir, pathExists, rmdir } from 'fs-extra';
import { join } from 'path';

import { Command } from '../../../src';
import { STORYBOOK_STORIES_MAP_PATH, STORYBOOK_UXPIN_CONFIG_PATH } from '../../../src/common/constants';
import { Environment } from '../../../src/program/env/Environment';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { testDirPath } from '../../utils/resources/testDirPath';

import { TEMP_DIR_PATH } from '../../../src/steps/building/config/getConfig';

const CURRENT_TIMEOUT:number = 150000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

const PROJECT_DIR:string = join(testDirPath, 'resources/repos/storybook-design-system');
const UXPIN_TEMP_DIR:string = join(PROJECT_DIR, TEMP_DIR_PATH);

describe('generates uxpin.config.js and/or componentsStoriesMap.js based on .storybook/main.js', () => {
  describe('With dump command', () => {

    beforeEach(async () => {
      if (await pathExists(UXPIN_TEMP_DIR)) {
        await emptyDir(UXPIN_TEMP_DIR);
        await rmdir(UXPIN_TEMP_DIR);
      }
    });

    describe('with --config option', () => {
      it('generates only componentsStoriesMap.js', async () => {
        await runUXPinMergeCommand({
          cwd: 'resources/repos/storybook-design-system',
          env: {
            UXPIN_ENV: Environment.TEST,
          },
          params: [
            Command.DUMP,
            '--storybook',
            '--config="./uxpin.config.js"',
          ],
        });

        const storiesMap:any = require(join(PROJECT_DIR, STORYBOOK_STORIES_MAP_PATH));
        expect(storiesMap).toMatchSnapshot();
        expect(await pathExists(join(PROJECT_DIR, STORYBOOK_UXPIN_CONFIG_PATH))).toBeFalsy();
      });
    });
    describe('without --config option', () => {
      it('generates both uxpin.config.js and componentsStoriesMap.js', async () => {
        await runUXPinMergeCommand({
          cwd: 'resources/repos/storybook-design-system',
          env: {
            UXPIN_ENV: Environment.TEST,
          },
          params: [
            Command.DUMP,
            '--storybook',
          ],
        });

        const uxpinConfig:any = require(join(PROJECT_DIR, STORYBOOK_UXPIN_CONFIG_PATH));
        const storiesMap:any = require(join(PROJECT_DIR, STORYBOOK_STORIES_MAP_PATH));
        expect(uxpinConfig).toMatchSnapshot();
        expect(storiesMap).toMatchSnapshot();
      });
    });
  });
});
