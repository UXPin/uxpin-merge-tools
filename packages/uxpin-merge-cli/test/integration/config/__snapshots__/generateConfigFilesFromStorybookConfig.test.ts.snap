// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates uxpin.config.js and/or componentsStoriesMap.js based on .storybook/main.js With dump command with --config option generates only componentsStoriesMap.js 1`] = `
Object {
  "Avatar": Object {
    "component": "Avatar",
    "componentFilePath": "./Avatar",
    "importDefault": false,
    "storiesPath": "src/components/Avatar.stories.js",
    "title": "Avatar",
  },
  "AvatarList": Object {
    "component": "AvatarList",
    "componentFilePath": "./AvatarList",
    "importDefault": false,
    "storiesPath": "src/components/AvatarList.stories.js",
    "title": "AvatarList",
  },
  "Badge": Object {
    "component": "Badge",
    "componentFilePath": "./Badge",
    "importDefault": false,
    "storiesPath": "src/components/Badge.stories.js",
    "title": "Badge",
  },
  "BulletLink": Object {
    "component": "BulletLink",
    "componentFilePath": "./BulletLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/BulletLink.stories.tsx",
    "title": "TableOfContents/BulletLink",
  },
  "Button": Object {
    "component": "Button",
    "componentFilePath": "./Button",
    "importDefault": false,
    "storiesPath": "src/components/Button.stories.js",
    "title": "Button",
  },
  "ButtonToggle": Object {
    "component": "ButtonToggle",
    "componentFilePath": "./ButtonToggle",
    "importDefault": false,
    "storiesPath": "src/components/ButtonToggle.stories.tsx",
    "title": "ButtonToggle",
  },
  "Cardinal": Object {
    "component": "Cardinal",
    "componentFilePath": "./Cardinal",
    "importDefault": false,
    "storiesPath": "src/components/Cardinal.stories.tsx",
    "title": "Cardinal",
  },
  "Checkbox": Object {
    "component": "Checkbox",
    "componentFilePath": "./Checkbox",
    "importDefault": false,
    "storiesPath": "src/components/Checkbox.stories.js",
    "title": "forms/Checkbox",
  },
  "Clipboard": Object {
    "component": "Clipboard",
    "componentFilePath": "./Clipboard",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/Clipboard.stories.tsx",
    "title": "Clipboard/Clipboard",
  },
  "ClipboardCode": Object {
    "component": "ClipboardCode",
    "componentFilePath": "./ClipboardCode",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardCode.stories.tsx",
    "title": "Clipboard/ClipboardCode",
  },
  "ClipboardIcon": Object {
    "component": "ClipboardIcon",
    "componentFilePath": "./ClipboardIcon",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardIcon.stories.tsx",
    "title": "Clipboard/ClipboardIcon",
  },
  "ClipboardInput": Object {
    "component": "ClipboardInput",
    "componentFilePath": "./ClipboardInput",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardInput.stories.tsx",
    "title": "Clipboard/ClipboardInput",
  },
  "CodeSnippets": Object {
    "component": "CodeSnippets",
    "componentFilePath": "./CodeSnippets",
    "importDefault": false,
    "storiesPath": "src/components/CodeSnippets.stories.js",
    "title": "CodeSnippets",
  },
  "FormErrorState": Object {
    "component": "FormErrorState",
    "componentFilePath": "./FormErrorState",
    "importDefault": false,
    "storiesPath": "src/components/FormErrorState.stories.tsx",
    "title": "forms/FormErrorState",
  },
  "Header": Object {
    "component": "Header",
    "componentFilePath": "./Header",
    "importDefault": false,
    "storiesPath": "src/components/header/Header.stories.js",
    "title": "Header",
  },
  "Highlight": Object {
    "component": "Highlight",
    "componentFilePath": "./Highlight",
    "importDefault": false,
    "storiesPath": "src/components/Highlight.stories.js",
    "title": "Highlight",
  },
  "Icon": Object {
    "component": "Icon",
    "componentFilePath": "./Icon",
    "importDefault": false,
    "storiesPath": "src/components/Icon.stories.js",
    "title": "Icon",
  },
  "Input": Object {
    "component": "Input",
    "componentFilePath": "./Input",
    "importDefault": false,
    "storiesPath": "src/components/Input.stories.js",
    "title": "forms/Input",
  },
  "ItemLink": Object {
    "component": "ItemLink",
    "componentFilePath": "./ItemLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/ItemLink.stories.tsx",
    "title": "TableOfContents/ItemLink",
  },
  "Link": Object {
    "component": "Link",
    "componentFilePath": "./Link",
    "importDefault": false,
    "storiesPath": "src/components/Link.stories.tsx",
    "title": "Link",
  },
  "LinkTabs": Object {
    "component": "LinkTabs",
    "componentFilePath": "./LinkTabs",
    "importDefault": false,
    "storiesPath": "src/components/LinkTabs.stories.js",
    "title": "LinkTabs",
  },
  "ListItem": Object {
    "component": "ListItem",
    "componentFilePath": "./ListItem",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/ListItem.stories.js",
    "title": "tooltip/ListItem",
  },
  "MenuLink": Object {
    "component": "MenuLink",
    "componentFilePath": "./MenuLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/MenuLink.stories.tsx",
    "title": "TableOfContents/MenuLink",
  },
  "Modal": Object {
    "component": "Modal",
    "componentFilePath": "./Modal",
    "importDefault": false,
    "storiesPath": "src/components/modal/Modal.stories.js",
    "title": "modal/Modal",
  },
  "ProgressDots": Object {
    "component": "ProgressDots",
    "componentFilePath": "./ProgressDots",
    "importDefault": false,
    "storiesPath": "src/components/ProgressDots.stories.js",
    "title": "ProgressDots",
  },
  "Radio": Object {
    "component": "Radio",
    "componentFilePath": "./Radio",
    "importDefault": false,
    "storiesPath": "src/components/Radio.stories.js",
    "title": "forms/Radio",
  },
  "Select": Object {
    "component": "Select",
    "componentFilePath": "./Select",
    "importDefault": false,
    "storiesPath": "src/components/Select.stories.js",
    "title": "forms/Select",
  },
  "ShadowBoxCTA": Object {
    "component": "ShadowBoxCTA",
    "componentFilePath": "./ShadowBoxCTA",
    "importDefault": false,
    "storiesPath": "src/components/ShadowBoxCTA.stories.js",
    "title": "ShadowBoxCTA",
  },
  "Spinner": Object {
    "component": "Spinner",
    "componentFilePath": "./Spinner",
    "importDefault": false,
    "storiesPath": "src/components/Spinner.stories.js",
    "title": "Spinner",
  },
  "Subheading": Object {
    "component": "Subheading",
    "componentFilePath": "./Subheading",
    "importDefault": false,
    "storiesPath": "src/components/Subheading.stories.js",
    "title": "Subheading",
  },
  "TableOfContents": Object {
    "component": "TableOfContents",
    "componentFilePath": "./TableOfContents",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/TableOfContents.stories.tsx",
    "title": "TableOfContents",
  },
  "TagItem": Object {
    "component": "TagItem",
    "componentFilePath": "./TagItem",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagItem.stories.tsx",
    "title": "Tag/TagItem",
  },
  "TagLink": Object {
    "component": "TagLink",
    "componentFilePath": "./TagLink",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagLink.stories.tsx",
    "title": "Tag/TagLink",
  },
  "TagList": Object {
    "component": "TagList",
    "componentFilePath": "./TagList",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagList.stories.tsx",
    "title": "Tag/TagList",
  },
  "Textarea": Object {
    "component": "Textarea",
    "componentFilePath": "./Textarea",
    "importDefault": false,
    "storiesPath": "src/components/Textarea.stories.js",
    "title": "forms/Textarea",
  },
  "Tooltip": Object {
    "component": "Tooltip",
    "componentFilePath": "./Tooltip",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/Tooltip.stories.js",
    "title": "tooltip/Tooltip",
  },
  "TooltipLinkList": Object {
    "component": "TooltipLinkList",
    "componentFilePath": "./TooltipLinkList",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipLinkList.stories.js",
    "title": "tooltip/TooltipLinkList",
  },
  "TooltipMessage": Object {
    "component": "TooltipMessage",
    "componentFilePath": "./TooltipMessage",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipMessage.stories.js",
    "title": "tooltip/TooltipMessage",
  },
  "TooltipNote": Object {
    "component": "TooltipNote",
    "componentFilePath": "./TooltipNote",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipNote.stories.js",
    "title": "tooltip/TooltipNote",
  },
  "WithModal": Object {
    "component": "WithModal",
    "componentFilePath": "./WithModal",
    "importDefault": true,
    "storiesPath": "src/components/modal/WithModal.stories.js",
    "title": "modal/WithModal",
  },
  "WithTooltip": Object {
    "component": "WithTooltip",
    "componentFilePath": "./WithTooltip",
    "importDefault": true,
    "storiesPath": "src/components/tooltip/WithTooltip.stories.js",
    "title": "tooltip/WithTooltip",
  },
}
`;

exports[`generates uxpin.config.js and/or componentsStoriesMap.js based on .storybook/main.js With dump command without --config option generates both uxpin.config.js and componentsStoriesMap.js 1`] = `
Object {
  "components": Object {
    "categories": Array [
      Object {
        "include": Array [
          "src/components/Avatar.tsx",
          "src/components/AvatarList.tsx",
          "src/components/Badge.js",
          "src/components/Button.js",
          "src/components/ButtonToggle.tsx",
          "src/components/Cardinal.tsx",
          "src/components/CodeSnippets.js",
          "src/components/Highlight.js",
          "src/components/Icon.tsx",
          "src/components/Link.tsx",
          "src/components/LinkTabs.js",
          "src/components/ProgressDots.js",
          "src/components/ShadowBoxCTA.js",
          "src/components/Spinner.js",
          "src/components/Subheading.js",
          "src/components/header/Header.tsx",
          "src/components/table-of-contents/TableOfContents.tsx",
        ],
        "name": "Uncategorized",
      },
      Object {
        "include": Array [
          "src/components/Checkbox.js",
          "src/components/FormErrorState.tsx",
          "src/components/Input.js",
          "src/components/Radio.js",
          "src/components/Select.js",
          "src/components/Textarea.js",
        ],
        "name": "forms",
      },
      Object {
        "include": Array [
          "src/components/clipboard/Clipboard.tsx",
          "src/components/clipboard/ClipboardCode.tsx",
          "src/components/clipboard/ClipboardIcon.tsx",
          "src/components/clipboard/ClipboardInput.tsx",
        ],
        "name": "Clipboard",
      },
      Object {
        "include": Array [
          "src/components/modal/Modal.js",
          "src/components/modal/WithModal.js",
        ],
        "name": "modal",
      },
      Object {
        "include": Array [
          "src/components/table-of-contents/BulletLink.tsx",
          "src/components/table-of-contents/ItemLink.tsx",
          "src/components/table-of-contents/MenuLink.tsx",
        ],
        "name": "TableOfContents",
      },
      Object {
        "include": Array [
          "src/components/tag/TagItem.tsx",
          "src/components/tag/TagLink.tsx",
          "src/components/tag/TagList.tsx",
        ],
        "name": "Tag",
      },
      Object {
        "include": Array [
          "src/components/tooltip/ListItem.js",
          "src/components/tooltip/Tooltip.js",
          "src/components/tooltip/TooltipLinkList.js",
          "src/components/tooltip/TooltipMessage.js",
          "src/components/tooltip/TooltipNote.js",
          "src/components/tooltip/WithTooltip.js",
        ],
        "name": "tooltip",
      },
    ],
  },
  "name": "Design System",
}
`;

exports[`generates uxpin.config.js and/or componentsStoriesMap.js based on .storybook/main.js With dump command without --config option generates both uxpin.config.js and componentsStoriesMap.js 2`] = `
Object {
  "Avatar": Object {
    "component": "Avatar",
    "componentFilePath": "./Avatar",
    "importDefault": false,
    "storiesPath": "src/components/Avatar.stories.js",
    "title": "Avatar",
  },
  "AvatarList": Object {
    "component": "AvatarList",
    "componentFilePath": "./AvatarList",
    "importDefault": false,
    "storiesPath": "src/components/AvatarList.stories.js",
    "title": "AvatarList",
  },
  "Badge": Object {
    "component": "Badge",
    "componentFilePath": "./Badge",
    "importDefault": false,
    "storiesPath": "src/components/Badge.stories.js",
    "title": "Badge",
  },
  "BulletLink": Object {
    "component": "BulletLink",
    "componentFilePath": "./BulletLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/BulletLink.stories.tsx",
    "title": "TableOfContents/BulletLink",
  },
  "Button": Object {
    "component": "Button",
    "componentFilePath": "./Button",
    "importDefault": false,
    "storiesPath": "src/components/Button.stories.js",
    "title": "Button",
  },
  "ButtonToggle": Object {
    "component": "ButtonToggle",
    "componentFilePath": "./ButtonToggle",
    "importDefault": false,
    "storiesPath": "src/components/ButtonToggle.stories.tsx",
    "title": "ButtonToggle",
  },
  "Cardinal": Object {
    "component": "Cardinal",
    "componentFilePath": "./Cardinal",
    "importDefault": false,
    "storiesPath": "src/components/Cardinal.stories.tsx",
    "title": "Cardinal",
  },
  "Checkbox": Object {
    "component": "Checkbox",
    "componentFilePath": "./Checkbox",
    "importDefault": false,
    "storiesPath": "src/components/Checkbox.stories.js",
    "title": "forms/Checkbox",
  },
  "Clipboard": Object {
    "component": "Clipboard",
    "componentFilePath": "./Clipboard",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/Clipboard.stories.tsx",
    "title": "Clipboard/Clipboard",
  },
  "ClipboardCode": Object {
    "component": "ClipboardCode",
    "componentFilePath": "./ClipboardCode",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardCode.stories.tsx",
    "title": "Clipboard/ClipboardCode",
  },
  "ClipboardIcon": Object {
    "component": "ClipboardIcon",
    "componentFilePath": "./ClipboardIcon",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardIcon.stories.tsx",
    "title": "Clipboard/ClipboardIcon",
  },
  "ClipboardInput": Object {
    "component": "ClipboardInput",
    "componentFilePath": "./ClipboardInput",
    "importDefault": false,
    "storiesPath": "src/components/clipboard/ClipboardInput.stories.tsx",
    "title": "Clipboard/ClipboardInput",
  },
  "CodeSnippets": Object {
    "component": "CodeSnippets",
    "componentFilePath": "./CodeSnippets",
    "importDefault": false,
    "storiesPath": "src/components/CodeSnippets.stories.js",
    "title": "CodeSnippets",
  },
  "FormErrorState": Object {
    "component": "FormErrorState",
    "componentFilePath": "./FormErrorState",
    "importDefault": false,
    "storiesPath": "src/components/FormErrorState.stories.tsx",
    "title": "forms/FormErrorState",
  },
  "Header": Object {
    "component": "Header",
    "componentFilePath": "./Header",
    "importDefault": false,
    "storiesPath": "src/components/header/Header.stories.js",
    "title": "Header",
  },
  "Highlight": Object {
    "component": "Highlight",
    "componentFilePath": "./Highlight",
    "importDefault": false,
    "storiesPath": "src/components/Highlight.stories.js",
    "title": "Highlight",
  },
  "Icon": Object {
    "component": "Icon",
    "componentFilePath": "./Icon",
    "importDefault": false,
    "storiesPath": "src/components/Icon.stories.js",
    "title": "Icon",
  },
  "Input": Object {
    "component": "Input",
    "componentFilePath": "./Input",
    "importDefault": false,
    "storiesPath": "src/components/Input.stories.js",
    "title": "forms/Input",
  },
  "ItemLink": Object {
    "component": "ItemLink",
    "componentFilePath": "./ItemLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/ItemLink.stories.tsx",
    "title": "TableOfContents/ItemLink",
  },
  "Link": Object {
    "component": "Link",
    "componentFilePath": "./Link",
    "importDefault": false,
    "storiesPath": "src/components/Link.stories.tsx",
    "title": "Link",
  },
  "LinkTabs": Object {
    "component": "LinkTabs",
    "componentFilePath": "./LinkTabs",
    "importDefault": false,
    "storiesPath": "src/components/LinkTabs.stories.js",
    "title": "LinkTabs",
  },
  "ListItem": Object {
    "component": "ListItem",
    "componentFilePath": "./ListItem",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/ListItem.stories.js",
    "title": "tooltip/ListItem",
  },
  "MenuLink": Object {
    "component": "MenuLink",
    "componentFilePath": "./MenuLink",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/MenuLink.stories.tsx",
    "title": "TableOfContents/MenuLink",
  },
  "Modal": Object {
    "component": "Modal",
    "componentFilePath": "./Modal",
    "importDefault": false,
    "storiesPath": "src/components/modal/Modal.stories.js",
    "title": "modal/Modal",
  },
  "ProgressDots": Object {
    "component": "ProgressDots",
    "componentFilePath": "./ProgressDots",
    "importDefault": false,
    "storiesPath": "src/components/ProgressDots.stories.js",
    "title": "ProgressDots",
  },
  "Radio": Object {
    "component": "Radio",
    "componentFilePath": "./Radio",
    "importDefault": false,
    "storiesPath": "src/components/Radio.stories.js",
    "title": "forms/Radio",
  },
  "Select": Object {
    "component": "Select",
    "componentFilePath": "./Select",
    "importDefault": false,
    "storiesPath": "src/components/Select.stories.js",
    "title": "forms/Select",
  },
  "ShadowBoxCTA": Object {
    "component": "ShadowBoxCTA",
    "componentFilePath": "./ShadowBoxCTA",
    "importDefault": false,
    "storiesPath": "src/components/ShadowBoxCTA.stories.js",
    "title": "ShadowBoxCTA",
  },
  "Spinner": Object {
    "component": "Spinner",
    "componentFilePath": "./Spinner",
    "importDefault": false,
    "storiesPath": "src/components/Spinner.stories.js",
    "title": "Spinner",
  },
  "Subheading": Object {
    "component": "Subheading",
    "componentFilePath": "./Subheading",
    "importDefault": false,
    "storiesPath": "src/components/Subheading.stories.js",
    "title": "Subheading",
  },
  "TableOfContents": Object {
    "component": "TableOfContents",
    "componentFilePath": "./TableOfContents",
    "importDefault": false,
    "storiesPath": "src/components/table-of-contents/TableOfContents.stories.tsx",
    "title": "TableOfContents",
  },
  "TagItem": Object {
    "component": "TagItem",
    "componentFilePath": "./TagItem",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagItem.stories.tsx",
    "title": "Tag/TagItem",
  },
  "TagLink": Object {
    "component": "TagLink",
    "componentFilePath": "./TagLink",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagLink.stories.tsx",
    "title": "Tag/TagLink",
  },
  "TagList": Object {
    "component": "TagList",
    "componentFilePath": "./TagList",
    "importDefault": false,
    "storiesPath": "src/components/tag/TagList.stories.tsx",
    "title": "Tag/TagList",
  },
  "Textarea": Object {
    "component": "Textarea",
    "componentFilePath": "./Textarea",
    "importDefault": false,
    "storiesPath": "src/components/Textarea.stories.js",
    "title": "forms/Textarea",
  },
  "Tooltip": Object {
    "component": "Tooltip",
    "componentFilePath": "./Tooltip",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/Tooltip.stories.js",
    "title": "tooltip/Tooltip",
  },
  "TooltipLinkList": Object {
    "component": "TooltipLinkList",
    "componentFilePath": "./TooltipLinkList",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipLinkList.stories.js",
    "title": "tooltip/TooltipLinkList",
  },
  "TooltipMessage": Object {
    "component": "TooltipMessage",
    "componentFilePath": "./TooltipMessage",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipMessage.stories.js",
    "title": "tooltip/TooltipMessage",
  },
  "TooltipNote": Object {
    "component": "TooltipNote",
    "componentFilePath": "./TooltipNote",
    "importDefault": false,
    "storiesPath": "src/components/tooltip/TooltipNote.stories.js",
    "title": "tooltip/TooltipNote",
  },
  "WithModal": Object {
    "component": "WithModal",
    "componentFilePath": "./WithModal",
    "importDefault": true,
    "storiesPath": "src/components/modal/WithModal.stories.js",
    "title": "modal/WithModal",
  },
  "WithTooltip": Object {
    "component": "WithTooltip",
    "componentFilePath": "./WithTooltip",
    "importDefault": true,
    "storiesPath": "src/components/tooltip/WithTooltip.stories.js",
    "title": "tooltip/WithTooltip",
  },
}
`;
